drivers/ethernet/eth_mcux.c

```diff
@@ -1029,14 +1029,18 @@ static int eth_init(const struct device *dev)
 
 #if defined(CONFIG_NET_IPV6)
 static void net_if_mcast_cb(struct net_if *iface,
-			    const struct in6_addr *addr,
+			    const struct net_addr *addr,
 			    bool is_joined)
 {
 	const struct device *dev = net_if_get_device(iface);
 	struct eth_context *context = dev->data;
 	struct net_eth_addr mac_addr;
 
-	net_eth_ipv6_mcast_to_mac_addr(addr, &mac_addr);
+	if (addr->family != AF_INET6) {
```

Comment:
I'd be good to move the Ethernet driver changes to a separate commit, they don't seem to be cover in the commit message (and are not related to the core IP stack changes).

Reply:
Done. Thanks!